smac/facade/smac_facade.py

```diff
@@ -207,75 +253,100 @@ def __init__(self,
 
         # initial EPM
         types, bounds = get_types(scenario.cs, scenario.feature_array)
+        model_def_kwargs = {
+            'types': types,
+            'bounds': bounds,
+            'instance_features': scenario.feature_array,
+            'seed': rng.randint(MAXINT),
+            'pca_components': scenario.PCA_DIM,
+        }
+        if model_kwargs is not None:
+            model_def_kwargs.update(model_kwargs)
         if model is None:
-            model = RandomForestWithInstances(types=types,
-                                              bounds=bounds,
-                                              instance_features=scenario.feature_array,
-                                              seed=rng.randint(MAXINT),
-                                              pca_components=scenario.PCA_DIM,
-                                              log_y=scenario.transform_y in ["LOG", "LOGS"],
-                                              num_trees=scenario.rf_num_trees,
-                                              do_bootstrapping=scenario.rf_do_bootstrapping,
-                                              ratio_features=scenario.rf_ratio_features,
-                                              min_samples_split=scenario.rf_min_samples_split,
-                                              min_samples_leaf=scenario.rf_min_samples_leaf,
-                                              max_depth=scenario.rf_max_depth)
+            for key, value in {
+                'log_y': scenario.transform_y in ["LOG", "LOGS"],
+                'num_trees': scenario.rf_num_trees,
+                'do_bootstrapping': scenario.rf_do_bootstrapping,
+                'ratio_features': scenario.rf_ratio_features,
+                'min_samples_split': scenario.rf_min_samples_split,
+                'min_samples_leaf': scenario.rf_min_samples_leaf,
+                'max_depth': scenario.rf_max_depth,
+            }.items():
+                if key not in model_def_kwargs:
+                    model_def_kwargs[key] = value
```

Comment:
That's looks more complicated than my solution with an `update()`. Why did you use a for loop here?

Reply:
If we would use update here we would overwrite arguments provided by the user which were added above in line 264. This is in line with adding a single argument as for example done in line 238/239 for the random configuration chooser.

Reply:
> If we would use update here we would overwrite arguments

That depends on how you apply the update. You could simply define your default arguments and overwrite them by the user provided arguments.

Reply:
Like:
```
           new_kwargs = {
                'log_y': scenario.transform_y in ["LOG", "LOGS"],
                'num_trees': scenario.rf_num_trees,
                'do_bootstrapping': scenario.rf_do_bootstrapping,
                'ratio_features': scenario.rf_ratio_features,
                'min_samples_split': scenario.rf_min_samples_split,
                'min_samples_leaf': scenario.rf_min_samples_leaf,
                'max_depth': scenario.rf_max_depth,
            }
            new_kwargs.update(model_def_kwargs)
            model_def_kwargs = new_kwargs
```
?